app/backends/workflow-orchestration/work-item-polling.ts

```diff
@@ -53,6 +53,50 @@ export async function getWorkFromDatabase(serviceID: string, reqLogger: Logger):
   return result;
 }
 
+/**
+ * Get work items from the database for the given service ID.
+ *
+ * @param serviceID - the id of the service to get work for
+ * @param reqLogger - a logger instance
+ * @param batchSize - batch size
+ * @returns work items from the database for the given service ID
+ */
+export async function getWorksFromDatabase(
+  serviceID: string,
+  reqLogger: Logger,
+  batchSize: number): Promise<WorkItemData[]> {
+  const workItems: WorkItemData[] = [];
+  try {
+    await db.transaction(async (tx) => {
+      const jobIds = await getNextJobIds(tx, serviceID as string, batchSize);
+      const workSize = (jobIds.length > 0) ? Math.ceil(batchSize / jobIds.length) : 1;
```

Comment:
This is intentional and a compromise between accuracy and speed. Since this outside get work loop is a busy loop in that it runs all the time without delay and by multiple schedulers, I have seen the get work getting called too many times rather than not enough. Although we could come up with a scenario where this function does not get the full batch of work items in one call, the other calls will pick up the slack and move the number of running pods higher. I have thought about this during implementation and didn't have a complete solution to always get the full batch size of work items in an efficient way and didn't think it is necessary, so I picked the current approach.

Reply:
Now, I think about it more and I think I should take your advice and try to add this adjustment since it is little work and could make ramping up the running pods faster. Even though this is not a complete solution, but it does make things better. Thanks!

Reply:
I updated the PR to adjust the workSize dynamically and also shuffle the jobIds to make it more robust. Really like Jame's suggestion.